astropy/io/fits/src/compressionmodule.c

```diff
@@ -1063,6 +1063,24 @@ PyObject* compression_decompress_hdu(PyObject* self, PyObject* args)
     if (PyErr_Occurred()) {
         return NULL;
     }
+    // non-integer scaling or subtractive dithering means floating-point
+    if (floor(fileptr->Fptr->cn_bscale) != fileptr->Fptr->cn_bscale ||
+        floor(fileptr->Fptr->cn_bzero) != fileptr->Fptr->cn_bzero ||
+        fileptr->Fptr->
+#ifdef CFITSIO_SUPPORTS_SUBTRACTIVE_DITHER_2
+            quantize_method
+#else
+            quantize_dither
+#endif
+                != NO_DITHER) {
+        if (datatype == BYTE_IMG || datatype == SHORT_IMG) {
```

Comment:
Is this some FITS convention? I could image a lot of (although artificial) cases where the result would be not representable with normal `float`s if I allow `BZERO` and `BSCALE` to be arbitary `double`s.

Reply:
I remember seeing the same conversion in the Python code, but I didn't find if it is described in the standard.

Reply:
There is https://github.com/astropy/astropy/blob/master/astropy/io/fits/hdu/image.py#L673 for instance, I was thinking of another piece of code but cannot find it right now.